e2e/app/component/dialog.ts

```diff
@@ -1,36 +1,31 @@
 import {ElementHandle, Page} from 'puppeteer';
+import Container from 'app/element/container';
 
 export enum ButtonLabel {
   Confirm = 'Confirm',
   KeepEditing = 'Keep Editing',
 }
 
-const SELECTOR = {
-  dialogRoot: '.ReactModal__Content[role="dialog"]',
-  button: '[role="button"]'
-}
 
-export default class Dialog {
-  private dialogElement: ElementHandle;
+export default class Dialog extends Container {
 
-  constructor(private readonly page: Page) {
+  private dialogElement: ElementHandle;
 
+  constructor(private readonly page: Page, selector: {xpath?: string, testId?: string}) {
+    super(selector);
   }
 
   async getContent(): Promise<string> {
     await this.findDialog();
-    const modalText = await this.page.evaluate((selector) => {
-      const modalElement = document.querySelector(selector);
-      return modalElement.innerText;
-    }, SELECTOR.dialogRoot);
-    console.log('dialog: ' + modalText);
-    return modalText;
+    const modalText = await (await this.dialogElement.getProperty('innerText')).jsonValue();
+    console.log('dialog: \n' + modalText);
+    return modalText.toString();
   }
 
   async clickButton(buttonLabel: ButtonLabel): Promise<void> {
     const selector = this.getButtonSelector();
-    await this.page.waitForSelector(selector, {visible: true});
-    const buttons = await this.page.$$(selector);
+    await this.page.waitForXPath(selector, {visible: true});
+    const buttons = await this.page.$x(selector);
```

Comment:
Could the selector be modified to include the text content or aria-label of the button? And if so would that eliminate the for loop and reduce some complexity?
I am seeing a 'findButton' function in xpath-finder.ts and it looks like there are text options that can be passed in - would this work?